cmd/admission-controller/mutatingwebhook.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/scylladb/go-log"
+	"github.com/scylladb/scylla-operator-autoscaler/pkg/api/v1alpha1"
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// recommendationApplier overwrites ScyllaCluster spec with recomendations given by Recommender (if available)
+type recommendationApplier struct {
+	Client  client.Client
+	decoder *admission.Decoder
+
+	logger log.Logger
+}
+
+func getDataCenterRecommendations(sca *v1alpha1.ScyllaClusterAutoscaler) []v1alpha1.DataCenterRecommendations {
+	if sca.Status.Recommendations == nil {
+		return nil
+	}
+	dcRecs := sca.Status.Recommendations.DataCenterRecommendations
+	if dcRecs == nil || len(dcRecs) == 0 {
+		return nil
+	}
+	return dcRecs
+}
+
+func getRackRecommendations(dataCenterName string,
+	dcRecs []v1alpha1.DataCenterRecommendations) []v1alpha1.RackRecommendations {
+	for idx := range dcRecs {
+		if dcRecs[idx].Name == dataCenterName {
+			if dcRecs[idx].RackRecommendations == nil {
+				return nil
+			}
+			return dcRecs[idx].RackRecommendations
+		}
+	}
+	return nil
+}
+
+func findRack(rackName string, racks []scyllav1.RackSpec) *scyllav1.RackSpec {
+	for idx := range racks {
+		if rackName == racks[idx].Name {
+			return &racks[idx]
+		}
+	}
+	return nil
+}
+
+func mutateCluster(ctx context.Context, logger log.Logger, cluster *scyllav1.ScyllaCluster) error {
+	logger.Info(ctx, "Starting mutation of ScyllaCluster")
+
+	c, err := client.New(config.GetConfigOrDie(), client.Options{Scheme: scheme})
+	if err != nil {
+		return fmt.Errorf("Failed to create a client: %s", err)
+	}
+
+	scas := &v1alpha1.ScyllaClusterAutoscalerList{}
+	err = c.List(ctx, scas)
+	if err != nil {
+		return fmt.Errorf("Failed to get SCAs: %s", err)
+	}
+
+	logger.Debug(ctx, "SCAs fetched", "num", len(scas.Items))
+
+	for idx := range scas.Items {
```

Comment:
Shouldn't this list be filetered to find SCA pointing to cluster under mutation?

Reply:
This comment is not resolved.